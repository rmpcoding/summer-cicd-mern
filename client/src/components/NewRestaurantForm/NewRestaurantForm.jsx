import React, { useState } from "react";

const NewRestaurantForm = () => {
  const [name, setName] = useState("");

  return (
    <form>
      <div className="mb-3">
        <label htmlFor="name" className="form-label">
          Name
        </label>
        <input
          type="text"
          className="form-control"
          id="name"
          data-testid="restaurant-name"
          name="name"
          value={name}
          onChange={(e) => setName(e.target.value)}
        />
      </div>
      <div className="text-center">
        <button type="submit" className="btn btn-primary">
          Submit
        </button>
      </div>
    </form>
  );
};

export default NewRestaurantForm;
