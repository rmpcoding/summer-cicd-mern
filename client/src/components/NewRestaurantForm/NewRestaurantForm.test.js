import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import NewRestaurantForm from "./NewRestaurantForm";

afterEach(cleanup);

describe("NewRestaurantForm", () => {
  it("updates the input value on change", () => {
    const { getByTestId } = render(<NewRestaurantForm />);
    const nameInput = getByTestId("restaurant-name");
    fireEvent.change(nameInput, { target: { value: "Waffle House" } });
    expect(nameInput.value).toBe("Waffle House");
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<NewRestaurantForm />);
    expect(asFragment()).toMatchSnapshot();
  });
});
