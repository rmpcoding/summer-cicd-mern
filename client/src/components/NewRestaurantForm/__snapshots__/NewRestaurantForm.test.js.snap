// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewRestaurantForm matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="name"
      >
        Name
      </label>
      <input
        class="form-control"
        data-testid="restaurant-name"
        id="name"
        name="name"
        type="text"
        value=""
      />
    </div>
    <div
      class="text-center"
    >
      <button
        class="btn btn-primary"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</DocumentFragment>
`;
